stl/inc/regex

```diff
@@ -3415,8 +3418,18 @@ bool _Matcher<_BidIt, _Elem, _RxTraits, _It>::_Better_match() { // check for bet
                 return _STD distance(_Begin, _Res._Grps[_Ix]._End) < _STD distance(_Begin, _Tgt_state._Grps[_Ix]._End);
             }
         }
+
+        if (_Res._Grp_valid[_Ix]) {
+            ++_Res_valid_count;
+        }
+
+        if (_Tgt_state._Grp_valid[_Ix]) {
+            ++_Tgt_state_valid_count;
+        }
     }
-    return false;
+
+    // more captured groups means a better match

```

Comment:
I have the following concerns:
1. Where is this mandated by ECMAScript? I am wary of adding logic to make this specific scenario work, without knowing exactly what Standardese it corresponds to.
2. Can we construct test cases that stress this rule (i.e. cases where it activates and causes different choices to be made), and compare the observed behavior of libc++, libstdc++, and browsers?
3. Does this affect the other grammars, and are they all specified to behave the same way?

In Boost's documentation (which is of course non-Standard, but more comprehensible than ECMAScript/POSIX/etc.) I see that the [Perl/ECMAScript behavior](https://www.boost.org/doc/libs/1_75_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html#boost_regex.syntax.perl_syntax.what_gets_matched) is described as "the first match found by a depth-first-search" and "leftmost", while [basic/extended behavior](https://www.boost.org/doc/libs/1_75_0/libs/regex/doc/html/boost_regex/syntax/leftmost_longest_rule.html) is "leftmost longest".